### DISCUSSION TITLE
beet not saving metadata after import?

### DISCUSSION THREAD

**Discussion by Normanras** [beetbox/beets#4761] [Q&A] [answered]
Before I submit a bug report, I wanted to check in with the community to make sure I'm doing this correct. This is my normal process for importing music. 

My album is downloaded to directory `artist/album/tracks`. I then run `beet import -c *` when in the `tracks` level of the directory or `beet import -c ./album` when in the album level. 

I go through all the prompts. First, it can't find the album in MusicBrainz, even though it clearly exists. I enter the ID in the prompts, confirm the tracks, etc. Great! 

After it's all done, I see that beet has also copied the files to `/musiclibrary/Non-Album/Artist/tracks`. Which is sort of close, but I don't know why it is going to the non-album folder. It should be `/musiclibrary/artist/album/tracks`, but that's a side issue. 

What is weird is that if I'm in my download dir or the `non-album` dir and I run `beet import` it says `No matching release found for 12 tracks`! The music also doesn't show up in my music player (Navidrome). Shouldn't beets be applying the metadata to all the tracks? Or is this expected behavior? 

If it's any help, [this is the musicbrainz album + id](https://musicbrainz.org/artist/dde55c12-c221-47a0-81f3-4101655dd394) I'm trying to import. 

Thanks in advance for your help!

**Comment by jackwilsdon** [comment]
Can you post your full configuration (`beet config`) and the output from a verbose import (`beet -vv import`)? Also it'd be great if you could post the tags on file files in the Non-Album directory using something like `ffprobe`.

Are you importing the album as a singleton somehow? That would explain how it has ended up in the `Non-Album` directory.

**Comment by Normanras** [comment]
```
directory: /media/Music/
library: /root/.config/beets/musiclibrary.db

import:
    copy: no
    write: no
    move: yes
    log: /root/.config/beets/import_2623.log

paths:
    default: $albumartist/$album%aunique{}/$track $title
```

Sure thing. I have `copy` as `no`, as I was under the impression that that would change the default functionality to not copy, but I could override it with the `-c` flag. Please correct me if that's wrong! 

And `beet -vv import Lost.opus` (just importing a single song)
```
user configuration: /root/.config/beets/config.yaml
data directory: /root/.config/beets
plugin paths:
Sending event: pluginload
library database: /root/.config/beets/musiclibrary.db
library directory: /media/Music
Sending event: library_opened
Sending event: import_begin
Sending event: import_task_created
Sending event: import_task_start
Looking up: /root/ytdl/Kings Kaleidoscope/Beyond Control/Lost.opus
Tagging  -
No album ID found.
Search terms:  -
Album might be VA: True
Evaluating 0 candidates.

/root/ytdl/Kings Kaleidoscope/Beyond Control/Lost.opus (1 items)
Sending event: before_choose_candidate
No matching release found for 1 tracks.
For help, see: http://beets.readthedocs.org/en/latest/faq.html#nomatch
[S]kip, Use as-is, as Tracks, Group albums, Enter search, enter Id, aBort?
```

This was the same prompt I got on my first attempt at importing - it didn't find any tracks and I had to manually search or enter the ID. 

Is this what you're looking for with `ffprobe`? Same as above, I just did it for a single file in the `Non-Album` directory.

```
Input #0, ogg, from 'Lost_.opus':
  Duration: 00:03:51.49, start: 0.007500, bitrate: 138 kb/s
    Stream #0:0(eng): Audio: opus, 48000 Hz, stereo, fltp
    Metadata:
      encoder         : Lavf58.45.100
```

**Reply by jackwilsdon** [reply]
You're right in thinking that `-c` overrides `import.copy`.

The reason beets can't find the album in MusicBrainz is that beets uses file tags by default to search MusicBrainz, but your files don't look to have any tags in them. You can try enabling the [`fromfilename`](https://docs.beets.io/en/stable/plugins/fromfilename.html) plugin if you'd like to try matching from filename instead.

Beets still can't find the album on MusicBrainz when re-importing the files from `Non-Album` because `import.write` set to `no`, which disables writing tags to the files (relating to the above regarding files without tags).

**Comment by Normanras** [comment]
Oh wow. I think I totally misunderstood the `write` key in the config then. Okay, this has been super helpful. I'm going to try a few things out tomorrow and see what progress I make. Appreciate you helping with this!

**Comment by Normanras** [comment, marked as answer]
Okay, @jackwilsdon, thank you! That worked. To give a synopsis of the above items needed in order to mark this as an answer, when downloading from sources that don't include any tags, using the plugin, [`fromfilename`](https://docs.beets.io/en/stable/plugins/fromfilename.html) will help `beets` locate the items from `MusicBrainz`. 

Second, in your config file, if you want beets to actually write the tags to the files, `import.write` needs to be set to yes, or omitted as the default is `yes`, [see here](https://beets.readthedocs.io/en/stable/reference/config.html#write).

Installing the plugin and changing the config has everything working as expected and I'm now enjoying the music! Thanks again!

**Comment by arsaboo** [comment]
`fromfilename` uses the file name to obtain tags, which can then be searched using any tag source.
